PocketKit/Sources/PocketKit/Home/CompactHomeCoordinator.swift

```diff
@@ -210,6 +211,17 @@ class CompactHomeCoordinator: NSObject {
 }
 
 extension CompactHomeCoordinator: UINavigationControllerDelegate {
+    func navigationController(_ navigationController: UINavigationController, willShow viewController: UIViewController, animated: Bool) {
+        // By default, when pushing the reader, switching to landscape, and popping,
+        // the list will remain in landscape despite only supporting portrait.
+        // We have to programatically force the device orientation back to portrait,
+        // if the view controller we want to show _only_ supports portrait
+        // (e.g when popping from the reader).
+        if viewController.supportedInterfaceOrientations == .portrait, UIDevice.current.orientation.isLandscape {
+            UIDevice.current.setValue(UIDeviceOrientation.portrait.rawValue, forKey: "orientation")
+        }
```

Comment:
This is unfortunate haha

What about subclassing UINavigationController and overriding `shouldAutorotate`? Would that get us anywhere?